TYPO3.Media/Classes/TYPO3/Media/ViewHelpers/ThumbnailViewHelper.php

```diff
@@ -0,0 +1,131 @@
+ * </code>
+ * <output>
+ * (depending on the image; scaled down to a width and height of 80 pixels, possibly changing aspect ratio)
+ * <img src="_Resources/Persistent/b29[...]95d.jpeg" width="80" height="80" alt="sample" />
+ * </output>
+ *
+ * <code title="Rendering an image with allowed up-scaling at given width and height">
+ * <typo3.media:image image="{imageObject}" maximumWidth="5000" allowUpScaling="true" alt="sample" />
+ * </code>
+ * <output>
+ * (depending on the image; scaled up or down to a width 5000 pixels, keeping aspect ratio)
+ * <img src="_Resources/Persistent/b29[...]95d.jpeg" width="80" height="80" alt="sample" />
+ * </output>
+ *
+ */
+class ThumbnailViewHelper extends AbstractTagBasedViewHelper
+{
+    /**
+     * @var \TYPO3\Flow\Resource\ResourceManager
+     * @Flow\Inject
+     */
+    protected $resourceManager;
+
+    /**
+     * @Flow\Inject
+     * @var \TYPO3\Media\Domain\Service\ThumbnailService
+     */
+    protected $thumbnailService;
+
+    /**
+     * @Flow\Inject
+     * @var \TYPO3\Media\Domain\Service\AssetService
+     */
+    protected $assetService;
+
+    /**
+     * name of the tag to be created by this view helper
+     *
+     * @var string
+     */
+    protected $tagName = 'img';
+
+    /**
+     * @return void
+     */
+    public function initializeArguments()
+    {
+        parent::initializeArguments();
+        $this->registerUniversalTagAttributes();
+        $this->registerTagAttribute('alt', 'string', 'Specifies an alternate text for an image', true);
+    }
+
+    /**
+     * Renders an HTML img tag with a thumbnail image, created from a given asset.
+     *
+     * @param AssetInterface $asset The asset to be rendered as an thumbnail
+     * @param integer $width Desired width of the image
+     * @param integer $maximumWidth Desired maximum width of the image
+     * @param integer $height Desired height of the image
+     * @param integer $maximumHeight Desired maximum height of the image
+     * @param boolean $allowCropping Whether the image should be cropped if the given sizes would hurt the aspect ratio
+     * @param boolean $allowUpScaling Whether the resulting image size might exceed the size of the original image
+     *
+     * @return string an <img...> html tag
+     */
+    public function render(AssetInterface $asset = null, $width = null, $maximumWidth = null, $height = null, $maximumHeight = null, $allowCropping = false, $allowUpScaling = false)
+    {
+        $thumbnailConfiguration = new ThumbnailConfiguration($width, $maximumWidth, $height, $maximumHeight, $allowCropping, $allowUpScaling);
+        $thumbnailData = $this->assetService->getThumbnailUriAndSizeForAsset($asset, $thumbnailConfiguration);
+
+        $this->tag->addAttributes(array(
+            'width' => $thumbnailData['width'],
+            'height' => $thumbnailData['height'],
+            'src' => $thumbnailData['src']
+        ));
+
+        return $this->tag->render();
+    }
+}
```

Comment:
not too happy with this additional view helper tbh., not sure it makes much sense to users.. the image view helper already accepts assets, although deprecated, but we could reverse that and deprecate the image argument instead like the asset argument is deprecated now and just use that one